ads/google/a4a/performance.js

```diff
@@ -138,11 +56,47 @@ export class BaseLifecycleReporter {
    * @param {string} unusedName A descriptive name for the beacon signal.
    */
   sendPing(unusedName) {}
+
+  /**
+   * Set a URL parameter to be added to the ping data.  The parameter's value is
+   * subject to URL replacement and both parameter name and value are URI
+   * encoded before being written to the ping.  The entry is silently dropped
+   * if either `parameter` or `value` is falsey, with the exception that the
+   * `value` may be 0.
+   *
+   * @param {string} parameter
+   * @param {string|number} value
+   */
+  setPingParameter(parameter, value) {
+    if (parameter == null || parameter === '') {
+      return;
+    }
+    if (value === null || value === undefined || value === '') { return; }
+    this.extraVariables_[parameter] = String(value);
```

Comment:
rather than coercing into a string, you could modify extraVariables_ to allow string|number values